### DISCUSSION TITLE
Birdseye hi-res camera

### DISCUSSION THREAD

**Discussion by Deishelon** [blakeblackshear/frigate#6894] [General]
Hi, 

My camera outputs hi-res & low-res stream, 
low-res stream is used for `detect` role, where hi-res is `record` role.

The camera config provides the `live->stream_name` (which is set to hi-res one) for good preview image. 
```
    live:
      stream_name: garage_front_main
```

However, when viewing birdseye stream - it takes the stream from low-res camera.
My birdseye config:
```
birdseye:
  enabled: True
  restream: True
  mode: continuous
```

Is there any way to make birdseye to use hi-res stream?
Or this is a limitation of the Birdseye mode? (In which case would be good to allow via config selecting the camera stream) 

Thanks! :)

**Comment by NickM-27** [comment]
The birdseye stream has to have each of the camera streams shown in it decoded. 

Decoding both the high and low res streams would be a very considerable amount of work that most devices I've seen people using couldn't keep up with.

**Reply by Deishelon** [reply]
Could it be made as an optional feature? Having an option is always good

**Reply by NickM-27** [reply]
The problem with an option is it has to actually work. It would not work for users with lightweight computers that are using the cameras substreams for detect (which is generally recommended). Users with powerful servers that could easily run high resolution birdseye with high res camera streams can already just run detect on the high res streams (or use their GPU to scale down the high res stream) anyway

**Reply by Inrego** [reply]
@NickM-27 you say that the option has to actually work. Why wouldn't it work? I'm aware that it probably wouldn't work on low-end hardware. But you can configure Frigate in a plethora of ways that low end hardware wouldn't be able to handle, so I don't think that's a valid argument.
And your "solution" to run detect on high res streams, is more of a workaround.. You didn't provide any real points to why it shouldn't be possible to run detect on low res (as is recommended), and birdseye on high res streams, except that it's just not a possibility at the moment. But I still think it's a valid feature request.

**Reply by NickM-27** [reply]
If you're running detect on 720p then having 4 cameras in birdseye you would need to be running birdseye configured at 2560x1440 for each stream to be at its native resolution. 

Now, if you run that same layout with each camera high res stream at 2560x1440 then you'd need to have birdseye set to 5120x2960 to have the cameras at native resolution. This would break birdseye restreaming because h264 has a max resolution of 4096 and very few devices have usable h265 encoders. 

So yes, it could supported but in most cases it would be pointless because if birdseye has more than 4 cameras having more than detect resolution wouldn't be using that higher resolution anyway.

**Reply by Inrego** [reply]
Does birdseye scale to keep original resolutions? I thought streams were downsized to fit into a fixed size stream for birdseye.
We're not asking for new magic to keep original resolution with variable stream resolution.
We're just asking to use the record stream as a source instead of detect stream. That way, the image could be clearer in birdseye (especially when it's only showing a single camera), and could be higher framerate (my detect stream is very low fps per recommendations)

**Reply by Inrego** [reply]
Edit, I want to clarify: We don't want native resolutions in birdseye as you mentioned. So it may just be a misunderstanding.
We just want to use the record stream instead of the detect stream. Frigate can still resize it down to whatever size it seems fit to fit everything into its target resolution. But using record streams as source would allow us to:

1. Have a clearer image (especially when only showing a single camera)
2. Show smoother video, since detect streams are often configured for very low fps (Frigate docs recommend 5 fps)

**Reply by NickM-27** [reply]
> Edit, I want to clarify: We don't want native resolutions in birdseye as you mentioned. So it may just be a misunderstanding.

I wasn't suggesting that's what you want but that's just how the math works out. If the streams are scaled down then there's no point in using a higher resolution because it's just creating more work for no benefit. When there's 4+ cameras being shown in birdseye it would not be beneficial.

> We just want to use the record stream instead of the detect stream. Frigate can still resize it down to whatever size it seems fit to fit everything into its target resolution. But using record streams as source would allow us to:
> 
> 1. Have a clearer image (especially when only showing a single camera)

Scaled down stream at 720p won't be clearer than a native 720p stream. So either way in most cases the quality of the birdseye stream would be the same or very similar regardless. Only when using 1 or 2 cameras would it be more clear and even then you'd have to be running birdseye at 4k for it to be more clear. 

> 2. Show smoother video, since detect streams are often configured for very low fps (Frigate docs recommend 5 fps)

To be clear Birdseye is capped at 10fps already due to performance reasons so I wouldn't see that changing with this anyway.
